dipy/reconst/mcsd.py

```diff
@@ -49,21 +59,27 @@ def multi_tissue_basis(gtab, sh_order, iso_comp):
 
 class MultiShellResponse(object):
 
-    def __init__(self, response, sh_order, shells):
+    def __init__(self, response, sh_order, shells, S0=None):
         """ Estimate Multi Shell response function for multiple tissues and
         multiple shells.
 
+        The method `multi_shell_fiber_response` allows to create a multi-shell
+        fiber response with the right format, for a three compartments model.
+        It can be refered to in order to understand the inputs of this class.
+
         Parameters
         ----------
-        response : tuple or AxSymShResponse object
-            A tuple with two elements. The first is the eigen-values as an (3,)
-            ndarray and the second is the signal value for the response
-            function without diffusion weighting.  This is to be able to
-            generate a single fiber synthetic signal.
+        response : ndarray
+            Multi-shell fiber response. The ordering of the responses should
+            follow the same logic as S0.
         sh_order : int
+            Maximal spherical harmonics order.
         shells : int
             Number of shells in the data
+        S0 : array (3,)
+            Signal with no diffusion weighting for each tissue compartments.
```

Comment:
In what order should these be provided? Maybe better for this to be a dict: `{"wm":value, "gm": value, "csf": value}`, or three input variables: `wm_S0`, `gm_S0`, `csf_S0`

Reply:
The latter would also help avoid the unfortunate clash with the S0 used by `predict` that you noticed below. 

Also: I am a bit confused. Where is this variable even being used?

Reply:
The order must be the same as the order of the responses. Since one could build a response with the order they want, I wanted to stay general.

Reply:
S0 is never used for now. I added it for a potential futur work on `predict`. I put comments in `predict` about ideas to used a S0!=1, but I was faced with the problem of needing the volume fractions or tissue masks. This could be useful is someone wants to predict with an arbitrary S0. In my case, I only see a use for the S0=1 case.

Reply:
OK - now I understand. I would add this to the docstring: "This is the S0 used for predicting from a fit model" or something like that. And what do you think about breaking this out into three different parameters?

Reply:
The problem is that one could want 4 tissues (for example), so I think we have to stay general.

Reply:
Gotcha. Yes. That makes sense. I think that expanding the docstring here would really help.

Reply:
I'll do that!

Reply:
Awesome. Don't forget to pull from your branch before moving ahead, as I've made some changes to it!